pkgs/build-support/go/module.nix

```diff
@@ -209,6 +221,8 @@ in
     ''
       runHook preBuild
 
+      export GOFLAGS="''${goFlags[*]}"
```

Comment:
Not sure if we'd want to do it, but we could have `goFlags` be appended to `GOFLAGS` in case it's already set, for some reason.

I'm trying to look at this from a standpoint where the build script eventually becomes a hook, which wouldn't be able to cleanly enforce using `goFlags` over `GOFLAGS`.

Reply:
> I'm trying to look at this from a standpoint where the build script eventually becomes a hook, which wouldn't be able to cleanly enforce using `goFlags` over `GOFLAGS`.

If the "hook" hear means a "setup hook", it can as long as it provides the `buildPhase` and the `checkPhase`.